File: lib/src/ui/chat/view_models/chat_event.dart
// Copyright (c) Yalochat, Inc. All rights reserved.

import 'package:equatable/equatable.dart';

sealed class ChatEvent {
  const ChatEvent();
}

final class ChatStartTyping extends ChatEvent with EquatableMixin {
  final String chatStatus;

  ChatStartTyping({this.chatStatus = ''});

  @override
  List<Object?> get props => [chatStatus];
}

final class ChatStopTyping extends ChatEvent with EquatableMixin {
  @override
  List<Object?> get props => [];
}

final class ChatUpdateUserMessage extends ChatEvent with EquatableMixin {
  final String value;

  const ChatUpdateUserMessage({required this.value});

  @override
  List<Object?> get props => [value];
}

final class ChatSendMessage extends ChatEvent with EquatableMixin {
  @override
  List<Object?> get props => [];
}

final class ChatClearMessages extends ChatEvent with EquatableMixin {
  @override
  List<Object?> get props => [];
}

File: android-sdk/sdk/src/main/kotlin/com/yalo/chat/sdk/domain/model/ChatEvent.kt
// Copyright (c) Yalochat, Inc. All rights reserved.

package com.yalo.chat.sdk.domain.model

// Port of flutter-sdk/lib/src/domain/models/chat_event/chat_event.dart
// Emitted by YaloMessageRepository.events() to drive the typing indicator lifecycle:
//   TypingStart — emitted when the user sends a message (agent is expected to reply)
//   TypingStop  — emitted when messages arrive from the server or on a fetch error
sealed class ChatEvent {
    data class TypingStart(val statusText: String) : ChatEvent()
    data object TypingStop : ChatEvent()
}
